Check mount paths exist

ManavalanG opened the pull request:
# Pull request

* Adds a verification step in the CLI wrapper script to check if the file/dirpaths to be mounted to singularity already exist as expected (#71). This would help debugging easier in cases where paths don't exist as expected.
* Resolves dirpaths of datasets in the workflow config file to obtain their full path

Please fill in the checklist below and comment as needed.

- [x] Was code modified? Briefly describe. _Yes. See description_.
- [x] Was documentation modified? Briefly describe. _No._
- [x] Is this a bug-fix? Briefly describe. _No._
- [x] Is this a feature addition? Briefly describe. _Yes. See description_.
- [x] Did you modify QuaC-Watch config file? If so, did you modify multiqc template
  `configs/multiqc_config_template.jinja2` and script `src/quac_watch/create_mutliqc_configs.py` as necessary? _No._
- [x] Did you perform system-level testing manually as described in master readme doc? Did it pass completely? If not why?
- [x] Updated `Changelog.md` file with change logs in recommended format?


## Anything else reviewer should know?

wilkb777 reviewed (changes requested):
I think it'd be better to change the `check_path_exists` function to take a list of paths and then check all entries in the list and return a list of missing paths all in one shot. Otherwise you'll have to do multiple runs of QuaC to catch all the missing paths with the current setup.

The extra benefit from this will also be that you can reduce the call of the function in `read_workflow_config` to a single call to check all the mount paths and files in one shot.

ManavalanG commented:
@wilkb777 Good point. I have now refactored to check mount paths all at once. Note that earlier I used to verify if the datasets specified in the workflow config file existed, but now the scope is restricted to just check the directories that will be mounted to singularity. That is, pipeline may still fail if the specified filepath doesn't exist under those directories. I think this is okay though.

Additionally, I have now refactored to resolve the dirpaths in the workflow config file; so no more partial paths. This should also take symlinks.

wilkb777 reviewed (approved):
Excellent, that's a great change along with the path existence assertion. LGTM!
